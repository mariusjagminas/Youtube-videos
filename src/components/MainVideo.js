import React from "react";
import { makeStyles } from "@material-ui/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";

const useStyle = makeStyles({
  container: {
    position: "relative",
    width: "100%",
    paddingBottom: "56.25%",
    overflow: "hidden"
  },
  iframe: {
    position: "absolute",
    top: 0,
    right: 0,
    width: "100%",
    height: "100%",
    border: 0
  }
});

const MainVideo = ({ currentVideo }) => {
  const classes = useStyle();

  return (
    <Card className={classes.card}>
      <div className={classes.container}>
        <iframe
          titlr={currentVideo.snippet.title}
          className={classes.iframe}
          src={`https://www.youtube.com/embed/${currentVideo.id.videoId}`}
          allowFullScreen={true}
        />
      </div>
      <CardContent>
        <Typography gutterBottom variant="h5" component="h2">
          {currentVideo.snippet.title}
        </Typography>
        <Typography variant="body2" color="textSecondary" component="p">
          {currentVideo.snippet.description}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default MainVideo;
