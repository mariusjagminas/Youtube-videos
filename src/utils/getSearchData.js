import axios from "axios";
import { config } from "../config";

const url = "https://www.googleapis.com/youtube/v3/search";

export const getSearchData = async term => {
  try {
    const response = await axios.get(url, {
      params: {
        part: "snippet",
        maxResults: 5,
        key: config.API_KEY,
        q: term
      }
    });
    return response.data.items;
  } catch {}
};

// https://www.youtube.com/embed/ video id
