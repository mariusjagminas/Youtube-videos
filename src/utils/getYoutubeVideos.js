import axios from "axios";

const url = "https://www.googleapis.com/youtube/v3/search";

export const getYoutubeVideos = async term => {
  try {
    const response = await axios.get(url, {
      params: {
        part: "snippet",
        maxResults: 5,
        key: process.env.REACT_APP_YOUTUBE_KEY,
        q: term
      }
    });
    // excluding playlists and channels items from returned list
    const videos = await response.data.items.filter(vid => vid.id.videoId);

    return { videos: videos, error: null };
  } catch (err) {
    return { videos: [], error: err.message };
  }
};
