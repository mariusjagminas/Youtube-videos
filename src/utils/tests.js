import renderer from "react-test-renderer";
import propTypesFunc from "check-prop-types";

export const checkIfMatchSnapshot = component => {
  return it("should match a snapshot", () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export const checkPropTypes = (component, props) => {
  return it("should have correct propTypes", () => {
    const result = propTypesFunc(
      component.propTypes,
      props,
      "props",
      component.name
    );
    expect(result).toBe(undefined);
  });
};
