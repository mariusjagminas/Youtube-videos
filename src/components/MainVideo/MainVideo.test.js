import React from "react";
import { checkIfMatchSnapshot } from "../../utils/tests";
import { shallow } from "enzyme";
import MainVideo from "./MainVideo";

describe("<MainVideo/>", () => {
  const props = {
    id: { videoId: "GHJ6899GHR" },
    snippet: {
      title: "test-title-83893HG",
      description: "test-description-JGH7688",
      publishedAt: "2018-12-10T14:13:40.000Z"
    }
  };

  checkIfMatchSnapshot(<MainVideo currentVideo={props} />);

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MainVideo currentVideo={props} />).dive();
  });

  it("should render <iframe/> element", () => {
    expect(
      wrapper.containsMatchingElement(
        <iframe src="https://www.youtube.com/embed/GHJ6899GHR" />
      )
    ).toBeTruthy();
  });

  it("should render the video title", () => {
    expect(wrapper.contains("test-title-83893HG")).toBeTruthy();
  });

  it("should render the video description ", () => {
    expect(wrapper.contains("test-description-JGH7688")).toBeTruthy();
  });

  it("should render date in correct format", () => {
    expect(wrapper.contains("Dec 10 2018")).toBeTruthy();
  });
});
