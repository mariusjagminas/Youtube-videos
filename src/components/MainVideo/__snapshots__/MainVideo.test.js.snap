// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainVideo component should match a snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
>
  <div
    className="MainVideo-container-1"
  >
    <iframe
      allowFullScreen={true}
      className="MainVideo-iframe-2"
      src="https://www.youtube.com/embed/GHJ6899GHR"
      title="test-title-83893HG"
    />
  </div>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      test-title-83893HG
    </h2>
    <h2
      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom"
      test-data="videoDate"
    >
      Dec 10 2018
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      test-description-JGH7688
    </p>
  </div>
</div>
`;
